src/Processors/Transforms/AggregatingTransform.cpp

```diff
@@ -687,11 +688,9 @@ void AggregatingTransform::consume(Chunk chunk)
 
 void AggregatingTransform::initGenerate()
 {
-    if (is_generate_initialized.load(std::memory_order_acquire))
+    if (is_generate_initialized.test_and_set())
```

Comment:
Is it the same as increasing memory order here?

Reply:
It's a stricter memory ordering so it is same with some possible performance implications
I don't think it should affect anything but if it does I will just set ordering to `memory_order_acq_rel`

Reply:
I'm not talking about performance, but rather why `atomic_flag` works here while `atomic<bool>` doesn't

Reply:
ah I see, `atomic_flag` is `atomic<bool>` with lock-free guarantee. The downside is the more limited API.
IMO it's good to use it if the `atomic<bool>` is just a simple flag that you read and set to `true` once because you get an extra guarantee for free.